### PR TITLE
Docs about tests collecting 

### PR DESCRIPTION
**Pull request by piotmag769**

### CODE REVIEW

File: docs/src/testing/debugging.md [hunk 1]
```diff
@@ -71,11 +71,6 @@ just `original value` is printed.
 > - Make sure to remove all of the prints before compiling the final version of your contract. 
 > - This feature is highly experimental and breaking changes are to be expected.
 
-To allow printing in tests you have to add the following line to your `Scarb.toml`:
-```toml
-[[target.starknet-contract]]
-allowed-libfuncs-deny = false
-```
```
Comment (piotmag769) [inline docs/src/testing/debugging.md:78]: Unrelated, but it was just not true, it is the default setting for starknet target in Scarb

File: docs/src/getting-started/scarb.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ Starknet Foundry uses [Scarb](https://docs.swmansion.com/scarb) to:
 
 One of the core concepts of Scarb is its [manifest file](https://docs.swmansion.com/scarb/docs/reference/manifest.html) - `Scarb.toml`.
 It can be also used to provide [configuration](../projects/configuration.md) for Starknet Foundry.
+Moreover, you can modify behaviour of `scarb test` to run `snforge` as 
+described [here](https://docs.swmansion.com/scarb/docs/extensions/testing.html#using-third-party-test-runners).
```
Comment (cptartur) [inline docs/src/getting-started/scarb.md:13]: Should it be a separate subsection on this page?

File: docs/src/testing/testing.md [hunk 3]
```diff
@@ -41,7 +41,88 @@ test case.
...
+├── tests/
+│   ├── lib.cairo
+│   ├── common/
+│   │   └── utils.cairo
+│   ├── common.cairo
+│   ├── test_contract.cairo
+│   └── not_included.cairo
+└── src/
+    └── lib.cairo
+```
```
Comment (cptartur) [inline docs/src/testing/testing.md:68]: Can we show a simpler example here? I get what are you trying to showcase, but I wonder if it isn't too confusing.

File: docs/src/testing/testing.md [hunk 4]
```diff
@@ -41,7 +41,88 @@ test case.
 Starknet Forge will collect tests only from these places:
 
 - any files reachable from the package root (declared as `mod` in `lib.cairo` or its children)
-- files inside the `tests` directory
+- files inside the [`tests`](#the-tests-directory) directory
+
+## The *tests* Directory
+
+### With *tests/lib.cairo*
```
Comment (cptartur) [inline docs/src/testing/testing.md:48]: I'd write something like: `Forge collects tests from `tests` directory. Depending on if `lib.cairo` file is defined there, the behavior of the test collector will be different.`

**Review by cptartur — CHANGES_REQUESTED** [review]

File: docs/src/getting-started/scarb.md (hunk 2 above)
Comment (piotmag769) [reply docs/src/getting-started/scarb.md:13]: I would leave it as it is

File: docs/src/testing/test-collection.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+and `tests/common.cairo` content:
+
+```rust
+mod utils;
+```
+
+tests from `tests/test_contract.cairo`, `tests/common.cairo` and `tests/common/utils.cairo` will be collected.
+
+### Common test modules
```
Comment (cptartur) [inline docs/src/testing/test-collection.md:88]: I'd name it something like `# Sharing Code Between Tests`

File: docs/src/testing/test-collection.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
+## Test collection
```
Comment (cptartur) [inline docs/src/testing/test-collection.md:1]: Let's capitalize the titles properly - `Test Collection` etc.

File: docs/src/SUMMARY.md [hunk 7]
```diff
@@ -21,6 +21,7 @@
 * [Using Cheatcodes](testing/using-cheatcodes.md)
 * [Debugging](testing/debugging.md)
 * [Fuzz Testing](testing/fuzz-testing.md)
+* [Test collection](testing/test-collection.md)
```
Comment (cptartur) [inline docs/src/SUMMARY.md:24]:
```suggestion
* [Test Collection](testing/test-collection.md)
```

**Review by cptartur — APPROVED** [review]

**Pull request merged**